File: src/app/models/todo.ts

export class Todo {
  id: number;
  title: string;
  priority: Priority;
  deadlineDate: any;
  deadlineTime: any;
  status: boolean;

  constructor(id: number, title: string, priority: Priority, deadlineDate: any, time: any, status: boolean) {
    this.id = id;
    this.title = title;
    this.priority = priority;
    this.deadlineDate = deadlineDate;
    this.deadlineTime = time;
    this.status = status;
  }
}

export enum Priority {
  Urgent,
  Important,
  Normal,
  NotUrgent
}

File: Models/Todo.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace VegaAPI.Models
{
    [Table("TodoItems")]
    public class Todo
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public Priority Priority { get; set; }
        public bool Status { get; set; }
        public DateTime DeadlineDate { get; set; }
        public DateTime DeadlineTime { get; set; }
    }
}